lib/internal/Magento/Framework/ObjectManager/Factory/AbstractFactory.php

```diff
@@ -104,11 +104,15 @@ public function getDefinitions()
      * @param array $args
      *
      * @return object
-     *
+     * @throws \Exception
      */
     protected function createObject($type, $args)
     {
-        return new $type(...array_values($args));
+        try {
+            return new $type(...array_values($args));
+        } catch (\Throwable $e) {
+            throw new \Exception($e->getMessage(), null, $e);
```

Comment:
I would suggest throwing `\Magento\Framework\Exception\RuntimeException` here instead, as `\Exception` is too generic.